src/Tools/SarifValidator/Validator.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.JSON.Core.Parser;
+using Microsoft.JSON.Core.Parser.TreeItems;
+using Microsoft.JSON.Core.Schema;
+using Microsoft.JSON.Core.Schema.Drafts.Draft4;
+
+namespace Microsoft.CodeAnalysis.Sarif.SarifValidator
+{
+    internal static class Validator
+    {
+        internal static IEnumerable<JSONError> ValidateFile(string instanceFilePath, string schemaFilePath)
+        {
+            string instanceText = File.ReadAllText(instanceFilePath);
+            string schemaText = File.ReadAllText(schemaFilePath);
+
+            return Validate(instanceText, schemaText);
+        }
+
+        private static IEnumerable<JSONError> Validate(string instanceText, string schemaText)
+        {
+            List<JSONError> errors = new List<JSONError>();
+
+            JSONDocument instanceDocument = JSONParser.Parse(instanceText);
+            AddSyntaxErrors(instanceDocument, JSONErrorLocation.InstanceDocument, errors);
+
+            JSONDocument schemaDocument = JSONParser.Parse(schemaText);
+            AddSyntaxErrors(schemaDocument, JSONErrorLocation.Schema, errors);
+
+            var loader = new JSONDocumentLoader();
+            loader.SetCacheItem(new JSONDocumentLoadResult(instanceDocument));
+            loader.SetCacheItem(new JSONDocumentLoadResult(schemaDocument));
+
+            JSONSchemaDraft4EvaluationTreeNode tree =
+                JSONSchemaDraft4EvaluationTreeProducer.CreateEvaluationTreeAsync(
+                        instanceDocument,
+                        (JSONObject)schemaDocument.TopLevelValue,
+                        loader,
+                        Enumerable.Empty<IJSONSchemaFormatHandler>()).Result;
+            AddValidationErrors(tree.ValidationIssues, errors);
+
+            return errors;
+        }
+
+        private static void AddSyntaxErrors(JSONDocument document, JSONErrorLocation location, List<JSONError> errors)
+        {
+            foreach (Tuple<JSONParseItem, JSONParseError> parserError in document.GetContainedParseErrors())
+            {
+                errors.Add(new JSONError
```

Comment:
Terrific! Then this tool's output can also be verified by the tool and now my brain hurts.